20241213_115526_trim_single_end/wdl/trimmomatic.wdl

```diff
@@ -0,0 +1,50 @@
+version 1.1
+
+task trim_single_end {
+    meta {
+        description: "Performs both single end and paired end trimming of reads using the Trimmomatic tool."
+        outputs: {
+            trimmed_reads: "The trimmed reads in FASTQ format."
+        }
+    }
+
+    parameter_meta {
+        input_file: "Input FASTQ format file to run Trimmomatic on."
+        output_name: "Output FASTQ format file of trimmed reads."
+        # adapter_file: "Adapter file to use for trimming."
+        leading: "Cut bases off the start of a read, if below a threshold quality."
+        trailing: "Cut bases off the end of a read, if below a threshold quality."
+        window_size: "Specifies the number of bases to average across for the sliding window."
+        window_quality: "Specifies the average quality required in the sliding window."
+        minlen: "Drop the read if it is below a specified length."
+    }
+
+    input {
+        File inputfile
+        String output_name
+        # String adapter_file
+        Int leading
+        Int trailing
+        Int window_size
+        Int window_quality
+        Int minlen
+    }
+
+    command <<<
+        cd /usr/local/share/trimmomatic-0.36-5
```

Comment:
Are you sure this is necessary? biocontainers _should_ have added a `trimmomatic` executable to the $PATH, as that's what they usually do. It should be unneeded to invoke the full path here.

Reply:
I was trying to work out access to the executable, but without cd'ing to the exact directory I wasn't able to get it to work

Reply:
That's likely because it's being called with `java`. There is a utility wrapper script at `/usr/local/bin/trimmomatic` that provides the functionality and is on `PATH`. So you should be able to swap `java -jar trimmomatic.jar` with `trimmomatic` and remove the directory change.

Reply:
Thank you @adthrasher! Hadn't noticed that distinction